### PR TITLE
Register suppliers - save new users in EMS DB

### PR DESCRIPTION
**Pull request by john-herciu-bjss**
Save generated user in the EMS database. Also added a new role (ROLE_SUPPLIER_ADMIN) with mostly ROLE_NHS permissions.

### CODE REVIEW

File: src/test/java/uk/nhs/ctp/service/UserManagementServiceTest.java [hunk 1]
```diff
@@ -31,7 +60,7 @@ public void createNewSupplierUser() {
     SupplierAccountDetails returned = userManagementService.createNewSupplierUser(request);
 
     SupplierAccountDetails expected = SupplierAccountDetails.builder()
-        .username("suppliedid")
+        .username("admin_suppliedid")
```
Comment (mhouston-nhs) [inline src/test/java/uk/nhs/ctp/service/UserManagementServiceTest.java:63]:
```suggestion
        .username("admin_supplierid")
```

Comment (john-herciu-bjss) [reply src/test/java/uk/nhs/ctp/service/UserManagementServiceTest.java]: Yep, changed

File: src/main/java/uk/nhs/ctp/controllers/UserController.java [hunk 2]
```diff
@@ -50,45 +50,45 @@
...
 
   @PreAuthorize("hasRole('ROLE_ADMIN')")
   @DeleteMapping(path = "/{username}")
   public @ResponseBody
-  void deleteUser(@PathVariable("username") String username) throws Exception {
+  void deleteUser(@PathVariable("username") String username) {
     userManagementService.deleteUser(username);
   }
 
   @PreAuthorize("hasRole('ROLE_ADMIN')")
```
Comment (cmarcus-nhs) [inline src/main/java/uk/nhs/ctp/controllers/UserController.java:81]: Not necessarily for this PR (but something we should capture) is the ability for a 'supplier admin' to create other users for that supplier.

File: src/test/java/uk/nhs/ctp/controllers/UserControllerTest.java [hunk 3]
```diff
@@ -24,16 +29,41 @@
...
   private UserController userController;
 
+  @Autowired
+  private UserRepository userRepository;
+
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
+  @After
+  public void cleanup() {
```
Comment (cmarcus-nhs) [inline src/test/java/uk/nhs/ctp/controllers/UserControllerTest.java:39]: this is ok but Spring provides 
`@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)`
Annotated on the class I think this is probably safer?

File: src/main/java/uk/nhs/ctp/controllers/UserController.java (hunk 2 above)
Comment (john-herciu-bjss) [reply src/main/java/uk/nhs/ctp/controllers/UserController.java:81]: Good point, raised [CDSCT-180](https://jira.digital.nhs.uk/browse/CDSCT-180) for this.

File: src/test/java/uk/nhs/ctp/controllers/UserControllerTest.java (hunk 3 above)
Comment (john-herciu-bjss) [reply src/test/java/uk/nhs/ctp/controllers/UserControllerTest.java:39]: That would have been great, but unfortunately couldn't get it to work. We need to look at not using the actual mysql database in the integration tests (couldn't easily get that working either).

**Pull request merged**